azmq/detail/context_ops.hpp

```diff
@@ -15,7 +15,8 @@
 #include <boost/assert.hpp>
 #include <boost/system/error_code.hpp>
 #include <boost/thread/mutex.hpp>
-#include <boost/thread/lock_guard.hpp>
+#include <boost/thread/locks.hpp>
```

Comment:
I generally try to keep boost includes to the narrowest possible header. Is lock_guard.hpp not in 1.48?

Reply:
lock_guard is inside locks.hpp in 1.48 and was split to separate header in 1.49.
And 1.48 does not contain mutex_type public typedef.
So, there are two ways - having single but more universal include locks.hpp or having additional conditional include.

Also, I suggest to retain 
`            static boost::mutex mtx;
`
declaration to simplify code instead of:
```
#if BOOST_VERSION < 104900
            static boost::mutex mtx;
#else
            static lock_type::mutex_type mtx;
#endif
```

Include of lock_guard in socker_service.hpp seems to be obsolete and could be removed.

I'm almost ready to squash these commits after resolving these questions.

Reply:
string_ref in socket_ops also optional, removing with conditional include.

Reply:
What do you prefer in context_ops.hpp?
```
-#include <boost/thread/lock_guard.hpp>
+#include <boost/thread/locks.hpp>
```
or
```
-#include <boost/thread/lock_guard.hpp>
+#include <boost/version.hpp>
+#if BOOST_VERSION >= 104900
+    #include <boost/thread/lock_guard.hpp>
+#else
+    #include <boost/thread/locks.hpp>
+#endif
```